src/Chapter4.hs

```diff
@@ -488,10 +494,29 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+instance Applicative List where
+   pure :: a -> List a
+   pure a = Cons a Empty
 
+   (<*>) :: List (a -> b) -> List a -> List b
+   fs <*> xs =  flatten (fmap (\x -> fmap (\f -> f x) fs) xs)
+   _ <*> _ = Empty
```

Comment:
Clever implementation! 🧠 And the last case can be removed because it's unreachable
```suggestion
   fs <*> xs =  flatten (fmap (\x -> fmap (\f -> f x) fs) xs)
```